plasmapy/particles/_factory.py

```diff
@@ -8,6 +8,7 @@
 
 import contextlib
 
+from numbers import Integral
```

Comment:
Yeah...I see your point.  We already require ``Z`` and ``mass_numb`` to be keyword-only, so that reduces the chances of mistakes but doesn't eliminate it.  One use case would be if you forgot the atomic number of an element, and want to do `element_name(118)` to look it up.  Another possibility would be if you wanted to do a loop over elements, like `[element_name(n) for n in range(1, 119)`, but that might be better served with a separate generator function like `element_range` that behaves similarly to `range`.  

Requiring an atomic number input to be keyword-only would be a non-trivial breaking change, so I'd rather preserve the existing behavior in the current PR.

Reply:
Makes sense - good subject for a future PR